### PR TITLE
Add support for custom readiness timeout seconds in istiod chart

### PR DESCRIPTION
**Pull request by Rez0k**
We faced an issue when the istio-proxy sidecar became unready when receiving a lot of requests, we want to increase the timeoutSeconds (readiness probe) to allow requests (even with a big delay) to pass to our microservice.

The current chart values file does not support adjusting the timeoutSeconds value of the readiness probe (adjusting with kustomize seems to be not easy and ugly), so I created this PR to allow devs like me to adjust the values to my needs.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ X] Configuration Infrastructure
- [ ] Docs
- [ ] Dual Stack
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by zirain** [comment]
/ok-to-test 

is there any related issue?

**Comment by Rez0k** [comment]
> /ok-to-test

We haven't opened an official issue, I think the timeoutSeconds should be adjustable, this is a need me and my team are now facing with the istio sidecar container.

File: operator/pkg/apis/istio/v1alpha1/values_types.proto [hunk 1]
```diff
@@ -969,15 +969,18 @@ message ProxyConfig {
   // Sets the number of successive failed probes before indicating readiness failure.
   uint32 readinessFailureThreshold = 22;
 
+  // Sets the number of seconds after which the probe times out
+	uint32 readinessTimeoutSeconds = 23;
```
Comment (zirain) [inline operator/pkg/apis/istio/v1alpha1/values_types.proto:973]: you can not do this, please use 40.

Comment (Rez0k) [reply operator/pkg/apis/istio/v1alpha1/values_types.proto]: done, can you help fix those tests?
I can't figure where I need to update the tests that fails

**Review by howardjohn — COMMENTED** [review]
This seems like a really bad idea. If the proxy isn't responding in 3s, its not ready -- its pretty much super broken? Are you sure you want to increase it?

BTW, it needs to fail *30 times* in a row to be unready

**Comment by howardjohn** [comment]
> > This seems like a really bad idea. If the proxy isn't responding in 3s, its not ready -- its pretty much super broken? Are you sure you want to increase it?
> > BTW, it needs to fail _30 times_ in a row to be unready
> 
> yes, its relevant especially for busy nodes (high CPU usage) or to edge cases for k8s scheduler This PR just adding this capability to change the timeout to the same value as the default , users should take their own responsibility for those changes

Its also our responsibility to not provide users with a way to cause outages easily. And to not introduce ever-growing complexity into the product by exposing 1000s of options that should *not* be configured, though.

**Comment by Rez0k** [comment]
> > > This seems like a really bad idea. If the proxy isn't responding in 3s, its not ready -- its pretty much super broken? Are you sure you want to increase it?
> > > BTW, it needs to fail _30 times_ in a row to be unready
> > 
> > 
> > yes, its relevant especially for busy nodes (high CPU usage) or to edge cases for k8s scheduler This PR just adding this capability to change the timeout to the same value as the default , users should take their own responsibility for those changes
> 
> Its also our responsibility to not provide users with a way to cause outages easily. And to not introduce ever-growing complexity into the product by exposing 1000s of options that should _not_ be configured, though.

so, you are saying this PR is not going to be approved, tnx for the response

**Pull request closed without merging**